### PR TITLE
Kanvi/bfloat16 tests

### PR DESCRIPTION
**Pull request by kanvi-nervana**
This PR implements the following new tests to test the bfloat16 conversions
1. test_fusedbatchnorm_training_nhwc.py
2. test_fusedbatchnorm_training_nchw.py
3. test_maxpoolbackprop_nhwc.py
4. test_maxpoolbackprop_nhwc.py

The following existing tests are also fixed as part of this PR
1. test_fusedbatchnorm.py
2. test_l2loss.py
3. test_maxpoolbackprop.py
4. test_relugrad.py
5. test_sparse_softmax_cross_entropy_with_logits.py

Also, made changes to the cmake so the new folder is included.

### CODE REVIEW

File: test/python/bfloat16/test_fusedbatchnorm_inference_nchw.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    x_t = tf.transpose(x_c, (0, 2, 3, 1))  # shape=[4, 1, 2, 3]
+
+    out_list = tf.nn.fused_batch_norm(
+        x_t,
+        scale,
+        offset,
+        mean,
+        variance,
+        data_format='NHWC',
+        is_training=False)
```
Comment (sayantan-nervana) [inline test/python/bfloat16/test_fusedbatchnorm_inference_nchw.py]: Is a test with `is_training=True` required?

Comment (sayantan-nervana) [reply test/python/bfloat16/test_fusedbatchnorm_inference_nchw.py]: NVM. just saw `test/python/bfloat16/test_fusedbatchnorm_training_nchw.py`

File: test/python/bfloat16/test_fusedbatchnorm_inference_nchw.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import pytest
+
+np.random.seed(5)
+
+# Inputs
+scale = [1.0, 0.9, 1.1]
+offset = [0.1, 0.2, -.3]
+mean = [0.4, 0.5, 0.6]
+variance = [0.1, 0.2, 0.3]
+input_shape = [4, 3, 1, 2]
```
Comment (shresthamalik) [inline test/python/bfloat16/test_fusedbatchnorm_inference_nchw.py]: name it input_shape_nchw

File: test/python/bfloat16/test_fusedbatchnorm_inference_nchw.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    with tf.Session(config=config) as sess_ng:
+        ngraph_bridge.enable()
+        ngraph_bridge.update_config(config)
+        os.environ['NGRAPH_TF_DISABLE_DEASSIGN_CLUSTERS'] = '1'
+        ng_out, in_0 = ng_model()
+        feed_dict = {in_0: k_np}
+        ng_outval = sess_ng.run(ng_out[0], feed_dict=feed_dict)
+
+    # transpose TF output from NHWC to NCHW for comparison with ngraph output
+    assert (np.allclose(
```
Comment (shresthamalik) [inline test/python/bfloat16/test_fusedbatchnorm_inference_nchw.py]: Do we care about only the 1st output of the op?

File: test/python/bfloat16/test_fusedbatchnorm_inference_nhwc.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    #Test 2: model2 with ngraph, NNP backend
+    with tf.Session(config=config) as sess_ng:
+        ngraph_bridge.enable()
+        ngraph_bridge.update_config(config)
+        os.environ['NGRAPH_TF_DISABLE_DEASSIGN_CLUSTERS'] = '1'
+        ng_out, in_0 = ng_model()
+        feed_dict = {in_0: k_np}
+        ng_outval = sess_ng.run(ng_out[0], feed_dict=feed_dict)
+
+    assert (np.allclose(tf_outval, ng_outval, rtol=0, atol=1e-02))
```
Comment (shresthamalik) [inline test/python/bfloat16/test_fusedbatchnorm_inference_nhwc.py]: Same, do we care only about the first output?

File: test/python/bfloat16/test_maxpoolbackprop_nchw.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+np.random.seed(5)
+
+#Inputs
+N = 4
+C = 3
+H = 8
+W = 8
+
+output_nchw = {
+    "VALID": np.random.rand(4, 3, 3, 3).astype('f'),
```
Comment (shresthamalik) [inline test/python/bfloat16/test_maxpoolbackprop_nchw.py]: can these shapes be declared as a variable and then reused

Comment (shresthamalik) [reply test/python/bfloat16/test_maxpoolbackprop_nchw.py]: https://github.com/tensorflow/ngraph-bridge/pull/183/files#diff-710d19f83aeab00a28522c3a83df1c8fR54

File: test/python/test_sparse_softmax_cross_entropy_with_logits.py [hunk 6]
```diff
@@ -44,17 +45,11 @@ def test_sparse_softmax_cross_entropy_with_logits_2d(self):
...
-            out = sparse_softmax_cross_entropy_with_logits(features, labels)
-            with self.session as sess:
-                result = sess.run(out)
+        expected = self.without_ngraph(sess_fn)
+        result = self.with_ngraph(sess_fn)
 
-        assert np.allclose(result[0], expected[0])
-        assert np.allclose(result[1], expected[1])
+        assert np.allclose(result[0], expected[0], rtol=0, atol=1e-02)
+        assert np.allclose(result[1], expected[1], rtol=0, atol=1e-02)
```
Comment (shresthamalik) [inline test/python/test_sparse_softmax_cross_entropy_with_logits.py]: does it not pass without the tolerance?

Comment (kanvi-nervana) [reply test/python/test_sparse_softmax_cross_entropy_with_logits.py]: Yes, it does not pass without tolerance.

File: test/python/bfloat16/test_maxpoolbackprop_nchw.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+stride_nchw = [1, 1, 2, 2]
+ksize_nchw = [1, 1, 3, 3]
+
+
+# TF graph
+def tf_model(padding):
+    orig_in = tf.placeholder(tf.float32, shape=[N, C, H, W])
+    if padding == "VALID":
+        grad = tf.placeholder(tf.float32, shape=[4, 3, 3, 3])
```
Comment (sindhu-nervana) [inline test/python/bfloat16/test_maxpoolbackprop_nchw.py]: you could use the variable and replace all these numbers for shapes etc.

File: test/python/bfloat16/test_maxpoolbackprop_nhwc.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+
+# TF graph
+def tf_model(padding):
+    orig_in = tf.placeholder(tf.float32, shape=[N, H, W, C])
+    if padding == "VALID":
+        grad = tf.placeholder(tf.float32, shape=[4, 3, 3, 3])
+        orig_out = tf.placeholder(tf.float32, shape=[4, 3, 3, 3])
+    elif padding == "SAME":
+        grad = tf.placeholder(tf.float32, shape=[4, 4, 4, 3])
```
Comment (sindhu-nervana) [inline test/python/bfloat16/test_maxpoolbackprop_nhwc.py]: same as above

File: test/python/bfloat16/test_fusedbatchnorm_training_nhwc.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+config = tf.ConfigProto(
+    allow_soft_placement=True,
+    log_device_placement=False,
+    inter_op_parallelism_threads=1)
+
+k_np = np.random.rand(4, 1, 2, 3).astype('f')  # NHWC
+
+
+def test_fusedbatchnorm_nchw():
```
Comment (sindhu-nervana) [inline test/python/bfloat16/test_fusedbatchnorm_training_nhwc.py]: test name should be nhwc

File: test/python/bfloat16/test_fusedbatchnorm_inference_nhwc.py [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+config = tf.ConfigProto(
+    allow_soft_placement=True,
+    log_device_placement=False,
+    inter_op_parallelism_threads=1)
+
+k_np = np.random.rand(4, 1, 2, 3).astype('f')  # NHWC
+
+
+def test_fusedbatchnorm_nchw():
```
Comment (sindhu-nervana) [inline test/python/bfloat16/test_fusedbatchnorm_inference_nhwc.py]: test name should be nhwc

File: test/python/bfloat16/test_fusedbatchnorm_training_nhwc.py [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
```
Comment (kanvi-nervana) [inline test/python/bfloat16/test_fusedbatchnorm_training_nhwc.py]:
```suggestion
def test_fusedbatchnorm_nhwc():
```

**Pull request merged**